C_CPP/Dynamic Memory allocation/Audio_for_dynamic_memory_allocation.md

```diff
@@ -0,0 +1,19 @@
+## DYNAMIC MEMORY ALLOCATION:-
+*The process of allocating memory to the variables during execution of the program or at run time is known as dynamic memory allocation.Till now whenever we needed an array
+we had decleared a static array of fixed size,say
+int arr[100];
+When this statement is executed,consecutive space for 100 integers is allocated .It is not uncommon that we may be using only 10% or 20% of the allocated space thereby wasting
+rest of the space .To overcome this problem and to utilize the memory efficiently,C language provides a mechanism of dynamically allocating memory so that only the amount
+of memory that is actually required is reserved.We reserve space only at the run time for the variables that are actually required.Dynamic memory allocation gives best
+performance in situations in which we do not know memory requirements in advance.
+C provides four library routines to automatically allocate memory at the run time.There are mainly 4 functions used in Dynamic memory allocation:-*
+
+- Malloc()
+- Calloc()
+- Free()
+- Realloc()
+<hr>
+
+### **Here is an audio giving overview of Dynamic memory allocation,its importance and use in C:-** ###
```

Comment:
Wrong level of heading, use ##.
No need for bold when using a heading.
No need of # at the end.
Remove ":-" from the end.